lib/location/method_gnss.c

```diff
@@ -866,6 +871,113 @@ static void method_gnss_print_pvt(const struct nrf_modem_gnss_pvt_data_frame *pv
 	}
 }
 
+#if defined(CONFIG_LOCATION_SERVICE_NRF_CLOUD_GNSS_POS_CLOUD_SEND)
+
+static int method_gnss_nrf_cloud_data_send(char *body)
+{
+	int ret;
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	struct nrf_cloud_tx_data mqtt_msg = {
+		.data.ptr = body,
+		.data.len = strlen(body),
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE,
+		.topic_type = NRF_CLOUD_TOPIC_MESSAGE,
+	};
+
+	ret = nrf_cloud_send(&mqtt_msg);
+	if (ret) {
+		LOG_ERR("MQTT: location data sending failed: %d", ret);
+	}
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	ret = nrf_cloud_coap_json_message_send(body, false, true);
```

Comment:
would be good if we could use cbor with the coap backend using `nrf_cloud_coap_location_send`.

Reply:
Seems we cannot encode the data into JSON string with CoAP then. I need to see how to do this at code level.

Reply:
Changed to `nrf_cloud_coap_location_send()` now.